pkg/tests/ossm/smcp_annotation_test.go

```diff
@@ -16,119 +16,119 @@ package ossm
 
 import (
 	_ "embed"
-	"strings"
+	"encoding/json"
+	"fmt"
 	"testing"
-	"time"
 
-	"github.com/maistra/maistra-test-tool/pkg/util"
 	"github.com/maistra/maistra-test-tool/pkg/util/env"
-	"github.com/maistra/maistra-test-tool/pkg/util/log"
+	"github.com/maistra/maistra-test-tool/pkg/util/hack"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/shell"
 	"github.com/maistra/maistra-test-tool/pkg/util/test"
+	. "github.com/maistra/maistra-test-tool/pkg/util/test"
 )
 
 var (
-	//go:embed yaml/deployment-testenv-x86.yaml
-	testenvDeploymentX86 string
-
-	//go:embed yaml/deployment-testenv-z.yaml
-	testenvDeploymentZ string
-
-	//go:embed yaml/deployment-testenv-p.yaml
-	testenvDeploymentP string
-
-	InjectedAnnotationsSMCPPatch = `
-spec:
-  proxy:
-    injection:
-      autoInject: true
-      injectedAnnotations:
-        test1.annotation-from-smcp: test1
-        test2.annotation-from-smcp: '["test2"]'
-        test3.annotation-from-smcp: '{test3}'
-`
+	InjectedAnnotationsSMCPPatch = `{"spec":{"proxy":{"injection":{"autoInject":true,"injectedAnnotations":{"test1.annotation-from-smcp":"test1","test2.annotation-from-smcp":"[\"test2\"]","test3.annotation-from-smcp":"{test3}"}}}}}`
 )
 
-func cleanupSMCPAnnotations() {
-	log.Log.Info("Cleanup ...")
-	util.KubeDeleteContents("bookinfo", testenvDeploymentX86)
-	time.Sleep(time.Duration(20) * time.Second)
-}
-
 func TestSMCPAnnotations(t *testing.T) {
-	test.NewTest(t).Id("T29").Groups(test.Full).NotRefactoredYet()
+	NewTest(t).Id("T29").Groups(Full).Run(func(t TestHelper) {
+		t.Log("Test annotations: verify deployment with sidecar.maistra.io/proxyEnv annotations and Enable automatic injection in SMCP")
```

Comment:
I don't know what this test does. I think it checks whether annotations are propagated properly from SMCP to the pod or something.

Reply:
Yes, I will expand the explanation in the log to:
`Test annotations: verify deployment with sidecar.maistra.io/proxyEnv annotations and Enable automatic injection in SMCP to propagate the annotations to the sidecar`